src/test/suite/lib/ggshield-configuration-utils.test.ts

```diff
@@ -0,0 +1,66 @@
+import * as simple from "simple-mock";
+import assert = require("assert");
+import { ExtensionContext, workspace } from "vscode";
+import * as GGShieldConfiguration from "../../../lib/ggshield-configuration";
+import { getConfiguration } from "../../../lib/ggshield-configuration-utils";
+
+suite("getConfiguration", () => {
+  let getConfigurationMock: simple.Stub<Function>;
+  let constructorMock: simple.Stub<Function>;
+  let context: Partial<ExtensionContext>;
+
+  setup(() => {
+    // Mock workspace.getConfiguration
+    getConfigurationMock = simple.mock(workspace, "getConfiguration");
+
+    // Mock the GGShieldConfiguration constructor globally
+    constructorMock = simple.mock(
+      GGShieldConfiguration,
+      "GGShieldConfiguration"
+    );
+    context = {} as ExtensionContext;
+  });
+
+  teardown(() => {
+    simple.restore();
+  });
+
+  test("Vscode settings are correctly read", async () => {
+    getConfigurationMock.returnWith({
+      get: (key: string) => {
+        if (key === "GGShieldPath") {
+          return "path/to/ggshield";
+        }
+        if (key === "apiUrl") {
+          return "https://custom-url.com";
+        }
+        if (key === "apiKey") {
+          return "test-api-key";
+        }
+        if (key === "allowSelfSigned") {
+          return true;
+        }
+      },
+    });
+    constructorMock.returnWith({});
+
+    getConfiguration(context as ExtensionContext);
```

Comment:
why can't we just do
```suggestion
    getConfiguration({} as ExtensionContext);
```